design/expected-hardware-configuration-validation.md

```diff
@@ -0,0 +1,262 @@
+
+        ```yaml
+        "metadata": {
+            "labels": {
+                "profile.metal3.io/profile1" : "BMO-OK",
+                "profile.metal3.io/profile2" : "BMO-OK"
+            }
+        }
+        ```
+
+
+* Create the Schema struct for `ExpectedHardwareConfiguration` inside `HardwareClassificationControllerSpec`,
+in file /api/v1alpha1/hardwareClassificationController_types.go.
+
+    ```yaml
+    type ExpectedHardwareConfiguration struct {
+        CustomFilter  string        `json:"customFilter"`
+        ProfileName   string        `json:"profileName"`
+        Namespace     string        `json:"namespace"`
+        Cpu           Cpu           `json:"CPU"`
+        Disk          Disk          `json:"Disk"`
+        Nics          Nics          `json:"NICS"`
+        Ram           int64         `json:"RAM"`
+        SystemVendor  SystemVendor  `json:"systemVendor"`
+        Firmware      Firmware      `json:"firmware"`
+    }
+    
+    type Cpu struct {
+        Count  int      `json:"count"`
+    }
+    
+    type Disk struct {
+        RootDisk       RootDisk    `json:"rootDisk"`
+        DataDisk       DataDisk    `json:"dataDisk"`
+    }
+
+    type RootDisk struct {
+        SizeBytesGB    int64      `json:"size"`
+    }
+
+    type DataDisk struct {
+        SizeBytesGB    int64     `json:"size"`
+        NumberOfDisks  int       `json:"numberOfDisks"`
+    }
+    
+    type Nics struct {
+        NumberOfNics   int      `json:"numberOfNics"`
+    }
+    
+    type SystemVendor struct {
+        ProductName     string  `json:"productName"`
+        Manufacturer    string  `json:"manufacturer"`
+    }
+
+    type Firmware struct {
+        Version Version `json:"version"`
+    }
+
+    type Version struct {
+        RAID               string `json:"RAID"`
+        BasebandManagement string `json:"baseBandManagement"`
+        BIOS               string `json:"BIOS"`
+        IDRAC              string `json:"IDRAC"`
+    }
+
+    type Rules struct {
+        Feild       string  `json:"feild"`
+        Operation   string  `json:"operation"`
+        Value       string  `json:"value"`
+    }
+    ```
+
+### Risks and Mitigations
+
+None
+
+## Design Details
+
+The following diagram shows the implementation details of Hardware Classification Controller.
+
+![Hardware Classification Controller Implementation Details](hwcc_sequence_diagram.png)
+
+
+### Work Items
+
+1. Implement CRD for `HardwareClassificationController`.
```

Comment:
This list of items is also still talking about working on lists of hosts instead of one host at a time.

Reply:
ok